### ISSUE TITLE
Awkward line breaks with long/compound types with type parameters

### ISSUE THREAD

**Issue by ceedubs** [scalameta/scalafmt#1133]
First off, thanks for such a great tool! After putting it on the back burner for a long, time, I'm back to trying to get scalafmt into Cats. The results are mostly great.

There's one formatting pattern that appears in several places and doesn't look great to me, and since I saw someone else reference it [here](https://gitter.im/scalameta/scalafmt?at=5a7b0edcb3c4a0d376f1e97c) on Gitter, I thought that I would go ahead and open an issue for it.

- **Version**: 1.3.0 (as dictated by sbt-plugin below)
- **Integration**: addSbtPlugin("com.lucidchart"      % "sbt-scalafmt"          % "1.15")

- **Configuration**:
```
maxColumn = 100

docstrings = JavaDoc

align = none
```

## Steps

Given code like this:
```scala
  implicit val catsDataInstancesForNonEmptyList: SemigroupK[NonEmptyList]
    with Reducible[NonEmptyList] with Bimonad[NonEmptyList] with NonEmptyTraverse[NonEmptyList] = ...

  implicit val catsDataInstancesForNonEmptyVector: SemigroupK[NonEmptyVector]
    with Reducible[NonEmptyVector] with Bimonad[NonEmptyVector]
    with NonEmptyTraverse[NonEmptyVector] = ...

   implicit val catsStdInstancesForOption
    : Traverse[Option] with MonadError[Option, Unit] with Alternative[Option]
      with CommutativeMonad[Option] with CoflatMap[Option] = ...
```

When I run scalafmt like this:

```bash
sbt scalafmt
```

## Problem

Scalafmt formats code like this (notice how type parameters wrap to the next line):
```scala

  implicit val catsDataInstancesForNonEmptyList: SemigroupK[NonEmptyList] with Reducible[
    NonEmptyList] with Bimonad[NonEmptyList] with NonEmptyTraverse[NonEmptyList] = ...

  implicit val catsDataInstancesForNonEmptyVector: SemigroupK[NonEmptyVector] with Reducible[
    NonEmptyVector] with Bimonad[NonEmptyVector] with NonEmptyTraverse[NonEmptyVector] = ...

   implicit val catsStdInstancesForOption
    : Traverse[Option] with MonadError[Option, Unit] with Alternative[Option] with CommutativeMonad[
      Option] with CoflatMap[Option] = ...
```

## Expectation

I would like the formatted output to look like this (same as input or similar):
```scala
  implicit val catsDataInstancesForNonEmptyList: SemigroupK[NonEmptyList]
    with Reducible[NonEmptyList] with Bimonad[NonEmptyList] with NonEmptyTraverse[NonEmptyList] = ...

  implicit val catsDataInstancesForNonEmptyVector: SemigroupK[NonEmptyVector]
    with Reducible[NonEmptyVector] with Bimonad[NonEmptyVector]
    with NonEmptyTraverse[NonEmptyVector] = ...

   implicit val catsStdInstancesForOption
    : Traverse[Option] with MonadError[Option, Unit] with Alternative[Option]
      with CommutativeMonad[Option] with CoflatMap[Option] = ...
```

## Workaround

I haven't found a workaround

**Comment by olafurpg** [comment]
Thanks for reporting! I've known about this problem for a while but it never bugged me since I rarely have such compound types in my own code. This seems like a duplicate of #1125 which is addressed in a PR opened yesterday #1132 . Could you please comment some of those examples in the PR so we can add them to the test suite?

**Comment by ceedubs** [comment]
@olafurpg wow, thanks for the extremely fast response. I'll add the examples there!
